DMCompiler/Compiler/DM/DMParser.cs

```diff
@@ -252,7 +252,10 @@ public DMASTStatement Statement(bool requireDelimiter = true)
                         statement = new DMASTObjectDefinition(loc, _currentPath, null);
                     }
 
-                    if (requireDelimiter && !PeekDelimiter() && Current().Type != TokenType.DM_Dedent) {
+                    var currentType = Current().Type;
```

Comment:
This isn't a proper fix, handling of the brackets should be left to the `BracedBlock()` method. A better fix may be to have that method end up calling `Statement()` with the `requireDelimiter` argument set to false since delimiters do not appear to be required in braced blocks.